### PR TITLE
Migrate projects list to MUI

### PR DESCRIPTION
**Pull request by ImanMahmoudinasab**
- [x] #401 
- [x] Describe the use-case in details
- [x] Ensure the solution is backwards-compatible
- [x] Test it. Submit screenshots / outputs / tests together with the PR.

image

image

image

image

image

### CODE REVIEW

File: packages/dashboard/src/project/projectList.tsx [hunk 1]
```diff
@@ -15,7 +17,7 @@ const ProjectsList = ({ search }: ProjectsListProps) => {
   const filters: Filters[] = search
     ? [
         {
-          value: null,
+          value: (undefined as unknown) as null,
```
Comment (ImanMahmoudinasab) [inline packages/dashboard/src/project/projectList.tsx:20]: This fixes an existing bug which cause searching projects not returning any result.

File: packages/dashboard/src/run/runDetails/resetInstance/resetInstanceButton.tsx [hunk 2]
```diff
@@ -24,10 +23,6 @@ export const ResetInstanceButton = ({
   instanceId: string;
   runId: string;
 }) => {
-  const {
-    params: { projectId },
-  } = useRouteMatch<{ projectId: string }>();
-
```
Comment (ImanMahmoudinasab) [inline packages/dashboard/src/run/runDetails/resetInstance/resetInstanceButton.tsx:30]: Removed unused codes to get rid of eslint errors.

File: packages/dashboard/src/project/projectList.tsx (hunk 1 above)
Comment (ImanMahmoudinasab) [reply packages/dashboard/src/project/projectList.tsx:20]: It is not the best way to solve the issue but the easiest. We should make this prop optional for `filters` as may someone need to pass `like` prop not the `value` prop.

**Review by agoldis — APPROVED** [review]
@ImanMahmoudinasab Loving it a lot! ❤️🎉

**Pull request merged**